src/Agent/NewRelic/Agent/Core/Configuration/DefaultConfiguration.cs

```diff
@@ -888,7 +882,10 @@ private bool IsDistributedTracingEnabled()
 
         public int? SamplingTarget => _serverConfiguration.SamplingTarget;
 
-        public int SpanEventsMaxSamplesStored => ServerOverrides(_serverConfiguration.EventHarvestConfig?.SpanEventHarvestLimit(), EnvironmentOverrides(_localConfiguration.spanEvents.maximumSamplesStored, "SPAN_EVENTS_MAX_SAMPLES_STORED").GetValueOrDefault()); public int? SamplingTargetPeriodInSeconds => _serverConfiguration.SamplingTargetPeriodInSeconds;
+        public int SpanEventsMaxSamplesStored => ServerOverrides(_serverConfiguration.SpanEventHarvestConfig?.HarvestLimit,
```

Comment:
This part is fine, but it might be worth to make sure that we always want to send up the non-negative user specified harvest limit for span events on the Connect.

https://github.com/newrelic/newrelic-dotnet-agent/blob/2ba99ea91ac20f79af40259e485949c6b3b1aac3/src/Agent/NewRelic/Agent/Core/DataTransport/ConnectModel.cs#L182

Reply:
Unfortunately there's no good way of accessing specifically the local configuration at this level, however there is no issue here - the `_serverConfiguration` used before Connect has a null `SpanEventHarvestConfig` so it always falls to the local config or default, and is then replaced by the server provided after Connect.

Reply:
You are correct. However, I'm specifically curious about what happens if users accidentally set the limit to `0` or a negative number. On startup, the agent will report this limit to the server then the server will response with the final limit. The server might already have a logic to handle this number but I'm not 100% sure. It is a rare case but it might be good if we could as well protect this from happen on the agent side.

Reply:
Just tested and negative values are kept and sent back by the server - that definitely doesn't make sense, so I can clamp the value here as you suggested.

Is zero not a valid value for this, though? Seems like it could be valid for some use cases, but I guess it could potentially break other things...

Reply:
per my conversation with other agent teams, we we seem to agree to enforce the minimum of 1000 for the following reasons:
1. Making sure DT is operated with minimal desired behavior. 1000 seems adequate, and historically customers wanted more.
2. Don’t have to deal with incorrect input .ie negative numbers.